### PR TITLE
refactored calendar component

### PR DESCRIPTION
**Pull request by mrclemds**

### CODE REVIEW

File: src/app/cras/edit-cra/calendar/calendar.component.ts [hunk 1]
```diff
@@ -142,77 +126,66 @@ export class CalendarComponent implements OnInit {
...
     this.emptyDays();
   }
 
-  previousMonth(date: string): void {
-    const lastMonth = subMonths(this[date], 1);
-
-    this[date] = lastMonth;
+  previousMonth(): void {
+    this.viewDate = subMonths(this.viewDate, 1);
     this.emptyDays();
```
Comment (RomainOliva) [inline src/app/cras/edit-cra/calendar/calendar.component.ts]: Why you don't do that in the `beforeMonthViewRender` method ?

File: src/app/cras/edit-cra/calendar/calendar.component.ts [hunk 2]
```diff
@@ -49,26 +32,28 @@ export class CalendarComponent implements OnInit {
...
+    const miniTimesheetKey = Object.keys(this.miniTimesheet)[0];
 
-    if (key !== undefined) {
-      const monthNumber = key.split('.')[0];
-      const year = key.split('.')[1];
-      const month = this.miniTimesheet[key];
+    if (miniTimesheetKey !== undefined) {
+      const monthNumber = miniTimesheetKey.split('.')[0];
+      const year = miniTimesheetKey.split('.')[1];
+      const month = this.miniTimesheet[miniTimesheetKey];
```
[the remark this answers is not in the document]
Comment (mrclemds) [reply src/app/cras/edit-cra/calendar/calendar.component.ts]: You're right

File: src/app/cras/edit-cra/calendar/calendar.component.ts [hunk 3]
```diff
@@ -78,58 +63,57 @@ export class CalendarComponent implements OnInit {
         if (month[date] !== undefined && month[date] !== 0) {
           this.addDay(day, addMinutes(day, month[date] * 60 * 8));
```
[the remark this answers is not in the document]
Comment (mrclemds) [reply src/app/cras/edit-cra/calendar/calendar.component.ts]: This allows me to have a day (8h) or half-day (4h) of work for the event and thus compute the 1 or the 0.5 shown on the calendar.

[the remark this answers is not in the document]
Comment (pfilippi) [reply src/app/cras/edit-cra/calendar/calendar.component.ts]: We should discuss this, but my instinct is telling me that this calendar component we are using might not be the best choice. We should consider implementing our own

**Pull request merged**